src/clients/KubectlClient.ts

```diff
@@ -101,10 +100,73 @@ export class KubectlClient implements IClient {
         }
     }
 
+    public async getContainersList(podName: string): Promise<string[]> {
+        if (!podName) {
+            this._logger.error(TelemetryEvent.KubectlClient_GetPodNameError, new Error(`Pod name is null`));
+            return null;
+        }
+        try {
+            // adding these decent amount of known sidecars to filter out the sidecars from the list
+            const knownSideCars: string[] = ['linkerd-proxy', 'linkerd-init', 'istio-proxy', 'darpd', 'jaeger-agent', 'nginx-proxy'];
+            const kubeconfigPath: string = await this._accountContextManager.getKubeconfigPathAsync();
+            const args: string[] = [`get`, `pod`, podName, `-o`, `jsonpath="{.spec['containers','initContainers'][*].name}"`];
+            const kubectlOutput: string = await this.runKubectlCommandAsync(args, kubeconfigPath);
+            if (kubectlOutput) {
+                // replace the double quotes and split the string by space ex: "linkerd-proxy stats-api linkerd-init" to [linkerd-proxy, stats-api, linkerd-init]
+                // and filter out the known sidecars from the list ex: [stats-api]
+                return kubectlOutput.replace('"', '').replace('"', '').split(' ')
+                    .filter(s => !knownSideCars.find(knownSideCar => knownSideCar == s));
+            } else {
+                this._logger.error(TelemetryEvent.KubectlClient_GetContainerListError, new Error(`container list is null`));
+                return null;
+            }
+        } catch (error) {
+            this._logger.error(TelemetryEvent.KubectlClient_GetContainerListError, error);
+            return null;
+        }
+
+    }
+
+    public async getPodName(serviceName: string): Promise<string> {
+        try {
+            const kubeconfigPath: string = await this._accountContextManager.getKubeconfigPathAsync();
+            // find the pod ipaddressList for the selected service
+            const args: string[] = [`get`, `ep`, serviceName, `-o`, `jsonpath='{.subsets[*].addresses[*].ip}'`];
+            let output = await this.runKubectlCommandAsync(args, kubeconfigPath);
+            // if there replicas and multiple pods, split it by apostrope, space example value: ''10.2.45.6 10.5.6.809''
+            // handles single value as well example value: ''10.56.78.90''
+            let ipaddressList: string[];
+            if (output && output.length > 0) {
+                ipaddressList = output.replace(/'/g, '').replace(/[\s,]+/g, ',').split(',');
+            }
+
+            if (ipaddressList?.length > 0) {
```

Comment:
I think we could skip this second loop altogether by including pod in the `jsonpath` expression:
```bash
$ kubectl get ep -n kube-system kube-dns -o jsonpath='{range .subsets[*].addresses[*]}{.ip}{" "}{.targetRef.name}{"\n"}{end}'
10.224.0.194 coredns-785fcf7bdd-4tw4c
10.224.0.217 coredns-785fcf7bdd-x7ltw
```

Actually, looking at it further, are the IP addresses not being included in the final output? If not, couldn't we just grab the pods directly, e.g.:
```bash
kubectl get ep -n kube-system kube-dns -o jsonpath='{.subsets[*].addresses[*].targetRef.name}'
```

(not sure if we'd need to filter by `kind==pod`)

Reply:
Thanks for suggestion, I missed this, this saved an extra call.